libraries/standard/mqtt/src/iot_mqtt_subscription.c

```diff
@@ -299,14 +299,19 @@ static bool _topicMatch( const IotLink_t * const pSubscriptionLink,
     const _mqttSubscription_t * pSubscription = IotLink_Container( _mqttSubscription_t,
                                                                    pSubscriptionLink,
                                                                    link );
-    const _topicMatchParams_t * pParam = ( _topicMatchParams_t * ) pMatch;
 
+    const _topicMatchParams_t * pParam = ( _topicMatchParams_t * ) pMatch;
+ 
     /* Extract the relevant strings and lengths from parameters. */
     const char * pTopicName = pParam->pTopicName;
     const char * pTopicFilter = pSubscription->pTopicFilter;
+    if ( pTopicName == NULL || pTopicFilter == NULL )
+    {
+        return status;
+    }
```

Comment:
Also check here before we pass them to `strncmp`. @gordonwang0

Reply:
The reason there's no NULL check here is because this is a private function (static) that expects its caller to perform parameter checks.  Can you move this check (and the one below) to the IsSubscribed function?

Reply:
I remove this code from `_topicMatch`. The check bellow is on the `isSubscribed` function and it'll be enough if you assume the caller will always perform the parameter check.